src/libraries/Common/src/System/Security/Cryptography/ECCng.ImportExport.cs

```diff
@@ -118,27 +118,44 @@ internal static byte[] GetPrimeCurveBlob(ref ECParameters parameters, bool ecdh)
 
         internal static void ExportNamedCurveParameters(ref ECParameters ecParams, byte[] ecBlob, bool includePrivateParameters)
         {
-            ECParameters temp = default;
-
-            DecodeEccKeyBlob(
-                ecBlob,
-                (KeyBlobMagicNumber magic, byte[] x, byte[] y, byte[]? d) =>
-                {
-                    CheckMagicValueOfKey(magic, includePrivateParameters);
-
-                    temp = new ECParameters
+            if (includePrivateParameters)
+            {
+                ecParams = DecodeEccKeyBlob(
+                    ecBlob,
+                    static (KeyBlobMagicNumber magic, byte[] x, byte[] y, byte[]? d) =>
                     {
-                        Q = new ECPoint
+                        CheckMagicValueOfKey(magic, includePrivateParameters: true);
+
+                        return new ECParameters
                         {
-                            X = x,
-                            Y = y,
-                        },
-                        D = d,
-                    };
-                },
-                clearPrivateKey: false); // Returning key to caller, so don't clear it.
+                            Q = new ECPoint
+                            {
+                                X = x,
+                                Y = y,
+                            },
+                            D = d,
+                        };
+                    },
+                    clearPrivateKey: false); // Returning key to caller, so don't clear it.
+            }
+            else
+            {
+                ecParams = DecodeEccKeyBlob(
+                    ecBlob,
+                    static (KeyBlobMagicNumber magic, byte[] x, byte[] y, byte[]? d) =>
+                    {
+                        CheckMagicValueOfKey(magic, includePrivateParameters: false);
```

Comment:
The way to let this be a static delegate while not duplicating the code would be to take a TState parameter in the callback.  The only state here is the boolean.

Of course, if other callers don't have state, then now you find yourself wanting to overload on with and without state (with the stateless driver just calling the stateful one with object:null)... so maybe that's more hassle than this duplicated code.  But maybe it's not.